[DO NOT MERGE] Test @wordpress/*@next packages

manzoorwanijk opened the pull request:
Testing the `@wordpress/*` packages as requested in https://github.com/WordPress/gutenberg/pull/73822#issuecomment-3681572502

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
*

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

manzoorwanijk committed:
Test @wordpress/*@next packages

manzoorwanijk committed:
Remove patches

manzoorwanijk committed:
Force is-shallow-equal to be the latest next version

jsnajdr committed:
Update to the Jan-7 next version

jsnajdr committed:
Update lockfile

manzoorwanijk committed:
Merge branch 'trunk' into test/wordpress-packages

manzoorwanijk commented:
CC @anomiex

anomiex committed:
Fix mangled pnpm lockfile

anomiex committed:
Add missing (and probably not actually needed) react dep

anomiex committed:
Add more hacking in pnpmfile to work around version issues upstream

anomiex committed:
Get some builds working

anomiex commented:
Looks likely that a lot of the failures are due to that situation. Several of the `@wordpress/*` packages don't seem to like it if multiple different versions are included in the same project.

I added some more hackery to get more of the builds working. Looks like https://github.com/WordPress/gutenberg/pull/73822 is still not fully-resolving references to external packages like `fast-deep-equal/es6` → `fast-deep-equal/es6/index.js` and `moment-timezone/moment-timezone-utils` → `moment-timezone/moment-timezone-utils.js`.

anomiex commented on projects/plugins/jetpack/tools/webpack.config.extensions.js:
```diff
@@ -290,7 +290,7 @@ module.exports = [
 						resource.request = path.join(
 							path.dirname( __dirname ),
-							'./extensions/shared/i18n-to-php'
+							'./extensions/shared/i18n-to-php.js'
```
Note this fix is legitimately needed in Jetpack.

anomiex committed:
Fix the hack for the hack

Various Calypso packages haven't been updated for the breaking change in `@wordpress/icons` v11.

jsnajdr committed:
Upgrade to Jan-7 next next version

jsnajdr committed:
Merge remote-tracking branch 'origin/trunk' into test/wordpress-packages

anomiex committed:
Cleanup hacks

This brings back non-next deps of various Calypso packages. If that's a problem, we could re-hack those.

anomiex committed:
Dataviews has a new missing dep

anomiex commented:
Looks like the E2Es are failing because dependency-extraction-webpack-plugin is extracting `@wordpress/ui` (which is used by `@wordpress/dataviews`) → `wp-ui` but nothing provides that handle, so the handles depending on it don't load.

jsnajdr committed:
Upgrade to Jan-9 next version (dewp)

jsnajdr committed:
Update patches

manzoorwanijk committed:
Fix @types/react version mismatch

manzoorwanijk committed:
Merge branch 'trunk' into test/wordpress-packages

manzoorwanijk commented on .pnpm-patches/@wordpress__components@32.0.1-next.ba3aee3a2.0.patch:
```diff
@@ -0,0 +1,34 @@
+ // packages/components/src/context/context-system-provider.js
+ import deepmerge from "deepmerge";
+-import fastDeepEqual from "fast-deep-equal/es6";
++import fastDeepEqual from "fast-deep-equal/es6/index.js";
```
Should we update the [imports in Gutenberg](https://github.com/search?q=repo%3AWordPress%2Fgutenberg+%22from+%27fast-deep-equal%2Fes6%27%22&type=code) to use the full path?

jsnajdr replied:
Yes, the Gutenberg imports should be updated, as they are not strictly compliant. The `fast-deep-equals` package doesn't have an `exports` field that would map the `es6` export to a specific file, so we should specify the full path. It's the same for the other packages.

Maybe it could be automated as part of the `transpilePackage` build step. Fully resolve not only local relative imports, as we already do, but also selected external imports.

manzoorwanijk replied:
Making it a part of the build process makes it more reliable.

manzoorwanijk commented on .pnpm-patches/@wordpress__compose@7.37.1-next.ba3aee3a2.0.patch:
```diff
@@ -0,0 +1,12 @@
+ // packages/compose/src/hooks/use-keyboard-shortcut/index.js
+ import Mousetrap from "mousetrap";
+-import "mousetrap/plugins/global-bind/mousetrap-global-bind";
++import "mousetrap/plugins/global-bind/mousetrap-global-bind.js";
```
Same for this import to update it upstream.

manzoorwanijk commented on .pnpm-patches/@wordpress__date@5.37.2-next.ba3aee3a2.0.patch:
```diff
@@ -0,0 +1,14 @@
+-import "moment-timezone/moment-timezone";
+-import "moment-timezone/moment-timezone-utils";
++import "moment-timezone/moment-timezone.js";
++import "moment-timezone/moment-timezone-utils.js";
```
Also for this

manzoorwanijk committed:
Fix one more @types/react mis-match

jsnajdr committed:
Remove patches

jsnajdr committed:
Upgrade to Jan-12 next version

jsnajdr committed:
Merge branch 'trunk' into test/wordpress-packages

jsnajdr committed:
Update forms and publicize dependencies

anomiex commented on .pnpmfile.cjs:
```diff
@@ -247,6 +253,15 @@ async function fixDeps( pkg ) {
 	}
 
+	// sass-embedded has an `unknown-all` and `all-unknown` optional dep that depends on `sass`.
+	// Let's remove these dependencies to prevent `sass` installation.
```
This is fine for this DNM PR, but once this comes up for real I'd recommend instead excluding the deps on these two packages entirely instead of neutering them like this.

anomiex replied:
Handled it in https://github.com/Automattic/jetpack/pull/46552.

anomiex committed:
Remove hopefully-unnecessary override

anomiex committed:
Remove all `-next.ba3aee3a2.0` versions

jsnajdr committed:
Merge branch 'trunk' into test/wordpress-packages

jsnajdr committed:
Upgrade to Jan14 next version

jsnajdr commented:
@anomiex @manzoorwanijk I updated this branch to the latest `.next.v.0` packages and the results are very good: just one e2e failure for Jetpack sync. Can you determine if that's just a flaky test or is something still broken?

manzoorwanijk committed:
Remove all @wordpress/* patch files

manzoorwanijk commented:
Yeah, the results are great. @anomiex Should we also see if we can clean up the `.pnpmfile.cjs` file?

anomiex committed:
Remove obsolete pnpmfile hack

anomiex committed:
Merge remote-tracking branch 'origin/trunk' into test/wordpress-packages

anomiex committed:
We can also remove setting `resolve.fullySpecified: false` for GB packages now
